tools/SourceKit/lib/SwiftLang/SwiftCompletion.cpp

```diff
@@ -649,22 +649,27 @@ void SwiftToSourceKitCompletionAdapter::getResultSourceText(
   auto Chunks = CCStr->getChunks();
   for (size_t i = 0; i < Chunks.size(); ++i) {
     auto &C = Chunks[i];
-    if (C.is(ChunkKind::BraceStmtWithCursor)) {
+    switch (C.getKind()) {
+    case ChunkKind::BraceStmtWithCursor:
       OS << " {\n" << getCodePlaceholder() << "\n}";
-      continue;
-    }
-    if (C.is(ChunkKind::CallParameterBegin)) {
+      break;
+    case ChunkKind::CallParameterBegin: {
       size_t Start = i++;
       for (; i < Chunks.size(); ++i) {
         if (Chunks[i].endsPreviousNestedGroup(C.getNestingLevel()))
           break;
       }
-      constructTextForCallParam(Chunks.slice(Start, i-Start), OS);
+      constructTextForCallParam(Chunks.slice(Start, i - Start), OS);
       --i;
-      continue;
+      break;
     }
-    if (!C.isAnnotation() && C.hasText()) {
+    case ChunkKind::OptionalMethodCallTail:
```

Comment:
Can we make the chunk not an annotation?  Then it would be handled by the simple case below.  Generally annotation chunks should be skipped.

Reply:
I kept it `isAnnotation()` to prevent `key.name` to have `!` in it. I'll try to modify `key.name` generation instead.

Reply:
Yeah, that would be better.  It wouldn't be the end of the world if the ! or ? ended up in the name  either.